CMakeLists.txt

```diff
@@ -0,0 +1,559 @@
+
+if(CMAKE_BUILD_TYPE_LOWER STREQUAL "release")
+    add_definitions(-DNDEBUG)
+endif()
+
+if(MSVC)
+    # Force to always compile with W4
+    if(CMAKE_C_FLAGS MATCHES "/W[0-4]")
+        string(REGEX REPLACE "/W[0-4]" "/W4" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+    else()
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /W4")
+    endif()
+elseif(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
+    # Update if necessary
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -pedantic")
+endif()
+
+# Disable bogus MSVC warnings
+if(MSVC)
+    add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+endif()
+
+# -fPIC thing
+if(NOT WIN32)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
+endif()
+
+# Prevent shared libraries has "lib" prefix on Windows DLL files
+if(WIN32)
+    set(CMAKE_SHARED_LIBRARY_PREFIX "")
+endif()
+# -----------------------------------
+
+
+if(POLICY CMP0077) # Cache variables override since 3.12
+    cmake_policy(SET CMP0077 NEW)
+endif()
+
+if(EMSCRIPTEN)
+    set(BUILD_SHARED_DEFAULT OFF)
+else()
+    set(BUILD_SHARED_DEFAULT ON)
+endif()
+
+option(BUILD_STATIC   "Build static library of libXMP" ON)
+option(BUILD_SHARED   "Build shared library of libXMP" ${BUILD_SHARED_DEFAULT})
+if(NOT BUILD_STATIC AND NOT BUILD_SHARED)
+    message(FATAL_ERROR "Both static and shared builds got disabled. You must enable at least one of them.")
+endif()
+
+option(LIBXMP_BUILD_LITE   "Build lite version of libXMP" ON)
+option(LIBXMP_BUILD_FULL   "Build full version of libXMP" ON)
+if(NOT LIBXMP_BUILD_LITE AND NOT LIBXMP_BUILD_FULL)
+    message(FATAL_ERROR "Both Full and Lite versions got disabled. You must enable at least one of them.")
+endif()
+
+option(LIBXMP_DISABLE_IT   "Disable IT support at libXMP-Lite" OFF)
+option(LIBXMP_DISABLE_DEPACKERS     "Disable archive depackers at libXMP" OFF)
+option(LIBXMP_DISABLE_PROWIZARD     "Disable the ProWizard at libXMP" OFF)
+
+TEST_BIG_ENDIAN(WORDS_BIGENDIAN)
+if(WORDS_BIGENDIAN)
+    add_definitions(-DWORDS_BIGENDIAN=1)
+endif()
+
+if(CMAKE_BUILD_TYPE_LOWER STREQUAL "debug")
+    add_definitions(-DDEBUG -D_DEBUG -DDEBUG_NOCOLOR)
+endif()
+
+check_symbol_exists(round "math.h;tgmath.h" HAVE_ROUND_FUNCTION)
+if(HAVE_ROUND_FUNCTION)
+    add_definitions(-DHAVE_ROUND=1)
+endif()
+
+check_symbol_exists(powf "math.h;tgmath.h" HAVE_POWF_FUNCTION)
+if(HAVE_POWF_FUNCTION)
+    add_definitions(-DHAVE_POWF=1)
+endif()
+
+if("${CMAKE_C_COMPILER_ID}" MATCHES "^(Apple)?Clang$")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-shift-negative-value")
+endif()
+
+if(MSVC)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /wd4244 /wd4018 /wd4996 /wd4048 /wd4267")
+    # set(CMAKE_LINK_LIBRARY_FLAG "${CMAKE_LINK_LIBRARY_FLAG} /wd4273")
+else()
+    if(NOT "${CMAKE_C_COMPILER_ID}" MATCHES "^(Apple)?Clang$")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-but-set-variable -Wno-stringop-truncation")
+    endif()
+    string(REGEX REPLACE "-pedantic" "" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+endif()
+
+set(CMAKE_C_STANDARD 99)
```

Comment:
Can't you use [target properties](https://cmake.org/cmake/help/latest/prop_tgt/C_STANDARD.html) for this? Modern CMake [really shouldn't](https://pabloariasal.github.io/2018/02/19/its-time-to-do-cmake-right/#targets-and-properties) use global stuff.

Reply:
Target properties are fine for same-language only, but they were a pain when you do use the mixture of C and C++ files in the same project. I do literally use this project at the sub-directory, and it doesn't conflict: all such options were local for the current branch, and that doesn't affect other projects. The only reason to affect another project when to `include()` this file.

Reply:
But wait... this something different... 
I do target into minimal CMake version 3.2 (keep the note that there are several Linux distros which have elder CMake at their packages, also count their older distro versions that keep elder CMake version)

https://cmake.org/cmake/help/v3.2/prop_tgt/C_STANDARD.html
Seems, that should work, lemme verify the thing...